server/src/test/java/io/crate/integrationtests/AlterTableIntegrationTest.java

```diff
@@ -59,4 +65,103 @@ public void test_alter_table_drop_column_dropped_meanwhile() {
+        execute("select * from t");
+        assertThat(response).hasRows("1| 11| 111");
+
+        execute("alter table t drop column a");
+        execute("select * from t");
+        assertThat(response).hasRows("1| 111");
+        Asserts.assertSQLError(() -> execute("select a from t"))
+            .hasPGError(UNDEFINED_COLUMN)
+            .hasHTTPError(NOT_FOUND, 4043)
+            .hasMessageContaining("Column a unknown");
+    }
+
+    @Test
+    public void test_alter_partitioned_table_drop_simple_column() {
+        execute("create table t(id integer primary key, a integer, b integer) partitioned by(id)");
+        execute("insert into t(id, a, b) values(1, 11, 111)");
+        execute("insert into t(id, a, b) values(2, 22, 222)");
+        execute("insert into t(id, a, b) values(3, 33, 333)");
+        execute("refresh table t");
+        execute("select * from t order by id");
+        assertThat(response).hasRows(
+            "1| 11| 111",
+            "2| 22| 222",
+            "3| 33| 333");
+
+        execute("alter table t drop column a");
+        execute("select * from t order by id");
+        assertThat(response).hasRows(
+            "1| 111",
+            "2| 222",
+            "3| 333");
+        execute("select * from t where id = 2");
+        assertThat(response).hasRows("2| 222");
+        Asserts.assertSQLError(() -> execute("select a from t"))
+            .hasPGError(UNDEFINED_COLUMN)
+            .hasHTTPError(NOT_FOUND, 4043)
+            .hasMessageContaining("Column a unknown");
+    }
+
+    @Test
+    public void test_alter_table_drop_leaf_subcolumn() {
+        // Use same names for sub-cols at different level of nesting
+        execute("create table t(id integer primary key, o object AS(a int, b int, oo object AS(a int, b int)))");
+        execute("insert into t(id, o) values(1, '{\"a\":1, \"b\":2, \"oo\":{\"a\":11, \"b\":22}}')");
+        execute("refresh table t");
+        execute("select * from t");
+        assertThat(response).hasRows("1| {a=1, b=2, oo={a=11, b=22}}");
+
+        execute("alter table t drop column o['oo']['b']");
+        execute("select * from t");
+        assertThat(response).hasRows("1| {a=1, b=2, oo={a=11}}");
+        Asserts.assertSQLError(() -> execute("select o['oo']['b'] from t"))
+            .hasPGError(UNDEFINED_COLUMN)
+            .hasHTTPError(NOT_FOUND, 4043)
+            .hasMessageContaining("Column o['oo']['b'] unknown");
+    }
+
+    @Test
+    public void test_alter_table_drop_subcolumn_with_children() {
+        // Use same names for sub-cols at different level of nesting
+        execute("create table t(id integer primary key, o object AS(a int, b int, oo object AS(a int, b int)))");
+        execute("insert into t(id, o) values(1, '{\"a\":1, \"b\":2, \"oo\":{\"a\":11, \"b\":22}}')");
+        execute("refresh table t");
+        execute("select * from t");
+        assertThat(response).hasRows("1| {a=1, b=2, oo={a=11, b=22}}");
```

Comment:
related to the comment on objects with nested arrays - could you add a test for that case? 
other way around - dropping array_of_something column would be also interesting

Reply:
yep true, thx, I had a note for this and I forgot. I'll open a followup for arrays if it's ok?

Reply:
> I'll open a followup for arrays 

sounds good.